index.html

```diff
@@ -0,0 +1,370 @@
+        {"State":"Montana","StateAbbr":"MT","felon":0.52,"vote":12,"Gerrymandering":"NA","PEI":4.51,"Restrictions":"Restrictions in Prison"},
+        {"State":"Oregon","StateAbbr":"OR","felon":0.5,"vote":"NA","Gerrymandering":"31.15","PEI":3.19,"Restrictions":"Restrictions in Prison"},
+        {"State":"Rhode Island","StateAbbr":"RI","felon":0.41,"vote":11,"Gerrymandering":"20.14","PEI":1.06,"Restrictions":"Restrictions in Prison"},
+        {"State":"Utah","StateAbbr":"UT","felon":0.4,"vote":10,"Gerrymandering":"27.66","PEI":0.34,"Restrictions":"Restrictions in Prison"},
+        {"State":"North Dakota","StateAbbr":"ND","felon":0.3,"vote":10,"Gerrymandering":"NA","PEI":-2.02,"Restrictions":"Restrictions in Prison"},
+        {"State":"New Hampshire","StateAbbr":"NH","felon":0.29,"vote":11,"Gerrymandering":"16.45","PEI":-2.36,"Restrictions":"Restrictions in Prison"},
+        {"State":"Massachusetts","StateAbbr":"MA","felon":0.25,"vote":7,"Gerrymandering":"18.45","PEI":0.19,"Restrictions":"Restrictions in Prison"},
+        {"State":"Maine","StateAbbr":"ME","felon":0,"vote":4,"Gerrymandering":"14.04","PEI":0.57,"Restrictions":"No Restrictions"},
+        {"State":"Vermont","StateAbbr":"VT","felon":0,"vote":2,"Gerrymandering":"NA","PEI":2.19,"Restrictions":"No Restrictions"}];
+
+      var MAX_DIS = 10.42;
+
+      var options = {
+        sorted: false,
+        filtered: false,
+        page: 0
+        };
+
+      var tbody = d3.select("tbody");
+
+      function updateTable() {
+        var data = states.slice();
+                     
+          // Erase any existing html code within tbody by replacing it with a blank string "".
+        tbody.html("");
+        // Binding data - this will be a very common task in our use of D3. Start by making a selectAll of <tr> elements that do not yet exist (they will soon!).
+        // Then we bind our data (schools) with .data and .enter()
+        // Lastly, we append the <tr> elements, which are become our current selection because of the prior selectAll("tr")
+      
+
+
+        if(options.filtered) {
+          if (options.filtered ==='no rest') {
+            data=data.filter(function (d) {return d.Restrictions==='No Restrictions'; });
+          } else if (options.filtered ==='prison') {
+            data=data.filter(function (d) {return d.Restrictions==='Restrictions in Prison'; });
+          } else if (options.filtered ==='parole') {
+            data=data.filter(function (d) {return d.Restrictions==='Restrictions While on Parole'; });
+          } else if (options.filtered ==='probation') {
+            data=data.filter(function (d) {return d.Restrictions==='Restrictions While on Probation'; });
+          } else if (options.filtered ==='post') {
+            data=data.filter(function (d) {return d.Restrictions==='Restrictions Post-Sentence'; });
+          }
+        }
```

Comment:
We'll want to avoid this pattern to the extent possible. For example, this might be better if we set the value of `options.filtered` directly to the value we want to check for in the data. If it is unavoidable, this might be a place to use a [switch statement](http://www.w3schools.com/js/js_switch.asp).